PostFreeSurfer/PostFreesurferPipelineLongPrep.sh

```diff
@@ -0,0 +1,433 @@
+function usage()
+{
+    #header text
+    echo "
+$log_ToolName: takes cross-sectional pre-Freesurfer output, as well as longitudinal FreeSurfer output folders.
+Creates volumes and transforms analogous to those produced by preFreesurferPipeline, but in longitudinal template space.
+
+Usage: $log_ToolName PARAMETER...
+
+PARAMETERs are [ ] = optional; < > = user supplied value
+"
+    #automatic argument descriptions
+    opts_ShowArguments
+    
+    #do not use exit, the parsing code takes care of it
+}
+
+#arguments to opts_Add*: switch, variable to set, name for inside of <> in help text, description, [default value if AddOptional], [compatibility flag, ...]
+#help info for option gets printed like "--foo=<$3> - $4"
+
+#TSC:should --path or --study-folder be the flag displayed by the usage?
+opts_AddMandatory '--path' 'StudyFolder' 'path' "folder containing all timepoins and templates" "--path"
+#The following options
+opts_AddMandatory '--subject'   'Subject'   'subject ID' "Subject label. Note: this is distinct subject label. Use separate labels for template and timepoints"
+opts_AddMandatory '--template'  'Template'  'FS template ID' "Longitudinal template ID (same as Freesurfer long template ID)"
+opts_AddMandatory '--timepoints' 'Timepoints_cross' 'FS timepoint ID(s)' "Freesurfer timepoint ID(s). For timepoint\
+                    processing, specify current timepoint. For template processing, must specify all timepoints, '@' separated"
+opts_AddMandatory '--template_processing' 'TemplateProcessing' 'create template flag' "0 if TP processing; 1 if template processing (must be run after all TP's)"
+
+#the following options have the same meaning as in PreFreesurferPipeline.
+opts_AddMandatory '--t1template' 'T1wTemplate' 'file_path' "MNI T1w template"
+opts_AddMandatory '--t1templatebrain' 'T1wTemplateBrain' 'file_path' "Brain extracted MNI T1wTemplate"
+opts_AddMandatory '--t1template2mm' 'T1wTemplate2mm' 'file_path' "MNI 2mm T1wTemplate"
+opts_AddMandatory '--t2template' 'T2wTemplate' 'file_path' "MNI T2w template"
+opts_AddMandatory '--t2templatebrain' 'T2wTemplateBrain' 'file_path' "Brain extracted MNI T2wTemplate"
+opts_AddMandatory '--t2template2mm' 'T2wTemplate2mm' 'file_path' "MNI 2mm T2wTemplate"
+opts_AddMandatory '--templatemask' 'TemplateMask' 'file_path' "Brain mask MNI Template"
+opts_AddMandatory '--template2mmmask' 'Template2mmMask' 'file_path' "Brain mask MNI 2mm Template"
+opts_AddMandatory '--fnirtconfig' 'FNIRTConfig' 'file_path' "FNIRT 2mm T1w Configuration file"
+
+#needed to transform Freesurfer mask to MNI space.
+opts_AddMandatory '--freesurferlabels' 'FreeSurferLabels' 'file' "location of FreeSurferAllLut.txt"
+opts_ParseArguments "$@"
+
+#display the parsed/default values
+opts_ShowValues
+
+# Naming Conventions
+T1wImage="T1w"
+T1wImageBrainMask="brainmask_fs"
+T1wFolder="T1w" #Location of T1w images
+T2wImage="T2w"
+T2wFolder="T2w" #Location of T2w images
```

Comment:
These variable names are longer than their contents, and thus are likely to only be confusing, please replace their uses in the code with their literal equivalents.  We generally only use `T1wFolder` to contain the entire absolute path to that folder, which does shorten commands and make the code easier to read, but that isn't what is being done here.  We generally don't put anything interesting into T2w/, and probably don't release it, either, so `T2wFolder` seems suspect to me anyway.

Reply:
These are a consistent with prior code and should be unchanged.

Reply:
PostFS has this:

https://github.com/Washington-University/HCPpipelines/blob/9cfa37a378dff26fd5aa4fd2aad24d0136830f0f/PostFreeSurfer/PostFreeSurferPipeline.sh#L161

This "longitudinal prep" script does not add the subject folder to that, and also, `$T1wFolder` doesn't actually get used in it, so at least the folder variables should be removed, as they would cause more harm than good (would get the wrong path if someone pasted in some code that used T1wFolder, which could be more complicated to discover than the "unbound variable" error from it not being defined).

The image name variables do get used here, but I doubt we plan to ever change them (and they are not variables in other pipelines), so they aren't doing anything useful even in PostFS.  Maybe the resulting increase in obviousness isn't important enough to actually remove them, or it is less important than other issues, but things would be more readable without variables with tiny values that aren't intended to ever vary.

Reply:
I suspect this is all copied straight from PreFreeSurfer.

Reply:
To clarify, the fact that `T1wFolder` currently gets set to just `"T1w"`, without the absolute path to get there, is NOT consistent with PostFS, and seems dangerous for any code that might get copied and pasted.  Even if we keep the silly, less-obvious `T1wImage` variable, we should do something to reduce the hazard of having an unused `T1wFolder` variable with an incomplete value in it.

Reply:
I kept these (TxwImage, TxwFolder) for styling compatibility, but if TxW images aren't frequently if ever renamed, following Tim's suggestion seems reasonable. The use of Txw/$TxwImage (same with *Folder) isn't consistent throughout structural scripts anyways. That would substantially simplify the code.

Reply:
Please remove the unused `T1wFolder` and `T2wFolder` variables, or at least include the missing `"$StudyFolder"/"$Subject"/` component that PostFS adds (if so, I recommend putting it all in a single obvious assignment like `T1wFolder="$StudyFolder"/"$Subject"/T1w`, unlike PostFS, to avoid this confusion going forward).  If you want to simplify other things while you are at it, I won't object.

Reply:
Ok. I also changed ${TXwImage} to TXw in that script.